pkg/controllers/user/cis/utils.go

```diff
@@ -107,3 +110,61 @@ func ValidateClusterBeforeLaunchingScan(cluster *v3.Cluster) error {
 	}
 	return nil
 }
+
+func GetBenchmarkVersionToUse(overrideBenchmarkVersion string, currentK8sVersion string,
+	cisConfigLister v3.CisConfigLister, cisConfigClient v3.CisConfigInterface,
+	cisBenchmarkVersionLister v3.CisBenchmarkVersionLister, cisBenchmarkVersionClient v3.CisBenchmarkVersionInterface,
+) (string, bool, error) {
+	bv := overrideBenchmarkVersion
+	shortK8sVersion := util.GetTagMajorVersion(currentK8sVersion)
+	if bv == "" {
+		cisConfigParams, err := kontainerdrivermetadata.GetCisConfigParams(
+			shortK8sVersion,
+			cisConfigLister,
+			cisConfigClient,
+		)
+		if err != nil {
+			logrus.Debugf("cisScanHandler: benchmark version not found for k8s version: %v, using default",
+				shortK8sVersion)
+			cisConfigParams, err = kontainerdrivermetadata.GetCisConfigParams(
+				"default",
+				cisConfigLister,
+				cisConfigClient,
+			)
+			if err != nil {
+				return "", false, fmt.Errorf("cisScanHandler: error fetching default cis config: %v", err)
+			}
+		}
+		bv = cisConfigParams.BenchmarkVersion
+	}
+	benchmarkInfo, err := kontainerdrivermetadata.GetCisBenchmarkVersionInfo(
+		bv,
+		cisBenchmarkVersionLister,
+		cisBenchmarkVersionClient,
+	)
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error fetching benchmark version info %v: %v",
+			bv, err)
+	}
+	benchmarkK8sVersionSemVer, err := util.StrToSemVer(ConvertToSemVerStr(benchmarkInfo.MinKubernetesVersion))
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error getting sem version for benchmark k8s version %v: %v",
+			benchmarkInfo.MinKubernetesVersion, err)
+	}
+	clusterK8sSemVerStr := ConvertToSemVerStr(shortK8sVersion)
+	clusterK8sSemVer, err := util.StrToSemVer(clusterK8sSemVerStr)
+	if err != nil {
+		return "", false, fmt.Errorf("cisScanHandler: error getting sem version for cluster k8s version %v: %v",
+			clusterK8sSemVer, err)
+	}
+	logrus.Debugf("cisScanHandler: checking if cluster version %v is less than min version: %v",
+		clusterK8sSemVerStr, benchmarkInfo.MinKubernetesVersion)
+	if clusterK8sSemVer.LessThan(*benchmarkK8sVersionSemVer) {
+		return "", false, fmt.Errorf("minimum k8s version %v needed for running cis scan", benchmarkInfo.MinKubernetesVersion)
+	}
```

Comment:
Can we to put the semver and version comparison in a separate method.

Reply:
👍